tests/gk-deploy/test_gk_deploy_basic.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+SCRIPT_DIR=$(cd $(dirname ${0}); pwd)
+STUBS_DIR="${SCRIPT_DIR}/stubs"
+TESTS_DIR="${SCRIPT_DIR}/.."
+INC_DIR="${TESTS_DIR}/common"
+BASE_DIR="${SCRIPT_DIR}/../.."
+DEPLOY_DIR="${BASE_DIR}/deploy"
+
+GK_DEPLOY="${DEPLOY_DIR}/gk-deploy"
+TOPOLOGY="${DEPLOY_DIR}/topology.json.sample"
+
+PATH="${STUBS_DIR}:$PATH"
+
+source "${INC_DIR}/subunit.sh"
+
+
+
+test_missing_topology () {
+	${GK_DEPLOY} -y
+	if [ "x$?" = "x0" ]; then
+		echo "ERROR: gk_deploy without toplogoy succeeded"
+		return 1
+	fi
+
+	return 0
+}
+
+test_cli_not_found () {
+	local saved_path="${PATH}"
+	PATH="/usr/bin:/bin"
+	local expected_out="Container platform CLI (e.g. kubectl, oc) not found."
+
+	OUT=$(${GK_DEPLOY} -y ${TOPOLOGY})
+	local rc=${?}
+
+	if [ "x${rc}" = "x0" ]; then
```

Comment:
This is a bash-ism. While we are using bash,
i still like to keep the bash-isms at a minimum and use posix-compatible alternatives where it does not disturb. similar for [[ vs [ . If more people want it the bash way, I can do that...